include/oneapi/dpl/internal/random_impl/exponential_distribution.h

```diff
@@ -0,0 +1,199 @@
+    operator()(_Engine& __engine, const param_type& __params)
+    {
+        return generate<size_of_type_, _Engine>(__engine, __params);
+    }
+
+    template <class _Engine>
+    result_type
+    operator()(_Engine& __engine, unsigned int __random_nums)
+    {
+        return operator()<_Engine>(__engine, param_type(lambda_), __random_nums);
+    }
+
+    template <class _Engine>
+    result_type
+    operator()(_Engine& __engine, const param_type& __params, unsigned int __random_nums)
+    {
+        return result_portion_internal<size_of_type_, _Engine>(__engine, __params, __random_nums);
+    }
+
+  private:
+    // Size of type
+    static constexpr int size_of_type_ = internal::type_traits_t<result_type>::num_elems;
+
+    // Static asserts
+    static_assert(::std::is_floating_point<scalar_type>::value,
+                  "oneapi::dpl::exponential_distribution. Error: unsupported data type");
+
+    // Distribution parameters
+    scalar_type lambda_;
+
+    // Implementation for generate function
+    template <int _Ndistr, class _Engine>
+    typename ::std::enable_if<(_Ndistr != 0), result_type>::type
+    generate(_Engine& __engine, const param_type __params)
+    {
+        return generate_vec<_Ndistr, _Engine>(__engine, __params);
+    }
+
+    // Specialization of the scalar generation
+    template <int _Ndistr, class _Engine>
+    typename ::std::enable_if<(_Ndistr == 0), result_type>::type
+    generate(_Engine& __engine, const param_type __params)
+    {
+        result_type __res;
+        uniform_real_distribution<scalar_type> __u;
+        __res = -std::log(scalar_type{1.0} - __u(__engine)) / __params.lambda;
+        return __res;
+    }
+
+    // Specialization of the vector generation  with size = [1; 3]
+    template <int __N, class _Engine>
+    typename ::std::enable_if<(__N <= 3), result_type>::type
+    generate_vec(_Engine& __engine, const param_type __params)
+    {
+        return generate_n_elems<_Engine>(__engine, __params, __N);
+    }
+
+    // Specialization of the vector generation with size = [4; 8; 16]
+    template <int __N, class _Engine>
+    typename ::std::enable_if<(__N > 3), result_type>::type
+    generate_vec(_Engine& __engine, const param_type __params)
+    {
+        uniform_real_distribution<result_type> __u;
+        result_type __res;
+        __res = __u(__engine);
+        __res = -sycl::log(scalar_type{1.0} - __res) / __params.lambda;
+        return __res;
+    }
+
+    // Implementation for the N vector's elements generation
+    template <class _Engine>
+    result_type
+    generate_n_elems(_Engine& __engine, const param_type __params, unsigned int __N)
+    {
+        result_type __res;
+        uniform_real_distribution __u;
+        for (int i = 0; i < __N; i++)
+        {
+            __res[i] = __u(__engine);
+            if (__res[i] < 0)
+                __res[i] = 0;
+            else
+                __res[i] = ((-1) / __params.lambda) * sycl::log(__res[i]);

```

Comment:
You use different equations in generate_vec and generate_n_elems. Please fix it

Reply:
done